docs/src/modules/utils/generateMarkdown.ts

```diff
@@ -298,6 +290,75 @@ The \`${reactAPI.styles.name}\` name can be used for providing [default props](/
 `;
 }
 
+/**
+ * Returns `null` if the prop should be ignored.
+ * Throws if it is invalid.
+ *
+ * @param prop
+ * @param propName
+ */
+function createDescribeableProp(
+  prop: PropDescriptor,
+  propName: string,
+): DescribeablePropDescriptor | null {
+  const { defaultValue, jsdocDefaultValue, description, required, type } = prop;
+
+  const renderedDefaultValue = defaultValue?.value.replace(/\r?\n/g, '');
+  const renderDefaultValue = Boolean(
+    renderedDefaultValue &&
+      // Ignore "large" default values that would break the table layout.
+      renderedDefaultValue.length <= 150,
+  );
+
+  if (description === undefined) {
+    throw new Error(`The "${propName}" prop is missing a description.`);
+  }
+
+  const annotation = doctrine.parse(description, {
+    sloppy: true,
+  });
+
+  if (
+    annotation.description.trim() === '' ||
+    annotation.tags.some((tag) => tag.title === 'ignore')
+  ) {
+    return null;
+  }
+
+  if (jsdocDefaultValue !== undefined && defaultValue === undefined) {
+    throw new Error(
+      `Declared a @default annotation in JSDOC for prop '${propName}' but could not find a default value in the implementation.`,
+    );
+  } else if (jsdocDefaultValue === undefined && defaultValue !== undefined && renderDefaultValue) {
+    const shouldHaveDefaultAnnotation =
+      // Discriminator for polymorphism which is not documented at the component level.
+      // The documentation of `component` does not know in which component it is used.
+      propName !== 'component';
+
+    if (shouldHaveDefaultAnnotation) {
+      // TODO: throw/warn/ignore?
+      // throw new Error(
+      //   `JSDOC @default annotation not found for '${propName}'.`,
+      // );
+      console.warn(`JSDOC @default annotation not found for '${propName}'.`);
```

Comment:
Why not throw here?

Reply:
Forgot to resolve

> // TODO: throw/warn/ignore?

We should definitely throw.

Reply:
It throws now.